auth/public/sdk/wapi.js

```diff
@@ -35,204 +34,93 @@ if (typeof wapiInit === "undefined") {
       const age = 3600 * 24 * 60;
       document.cookie = `token=${wapi.token};Secure;path=/;max-age=${age};`;
     };
+
     //scrub the api keys from wapi and deletes it from cookies
     wapi.scrubToken = function () {
       document.cookie = "token=;max-age=-1;path=/;";
       wapi.token = null;
     };
 
     //opens up a child window for auth stuff
-    wapi.openAuthPortal = () =>
-      (wapi.childWindow = window.open(authUrl, "_blank"));
+    wapi.openAuthPortal = () => wapi.childWindow = window.open(authUrl, "_blank");
 
     //checks if wapi is currently signed in
     wapi.isSignedIn = () => wapi.token != null;
+
     //signs out wapi
     wapi.signOut = () => wapi.scrubToken();
 
     //listens for an oauth result from the child window
-    wapi.authListen = function (setAuth) {
-      window.addEventListener("message", function (e) {
-        if (e.data.type === "auth") {
-          wapi.setToken(e.data.token);
-          if (setAuth != null) setAuth(wapi.isSignedIn());
-        }
-      });
-    };
+    wapi.authListen = setAuth => window.addEventListener("message", e => {
+      if (e.data.type === "auth") {
+        wapi.setToken(e.data.token || "");
```

Comment:
This was setting the token to `"undefined"` and `wapi.readToken = () => !wapi.token` returns `true`